test/rest-coercion/jsonform-integer.suite.js

```diff
@@ -0,0 +1,75 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: strong-remoting
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var jsonFormContext = require('./_jsonform.context');
+
+module.exports = function(ctx) {
+  ctx = jsonFormContext(ctx);
+  var EMPTY_BODY = ctx.EMPTY_BODY;
+  var ERROR_BAD_REQUEST = ctx.ERROR_BAD_REQUEST;
+  var verifyTestCases = ctx.verifyTestCases;
+
+  describe('json form - integer - required', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'integer', required: true }, [
+      // Valid values
+      [{ arg: 0 }, 0],
+      [{ arg: 1 }, 1],
+      [{ arg: -1 }, -1],
+
+      // Empty values should trigger ERROR_BAD_REQUEST
+      [EMPTY_BODY, ERROR_BAD_REQUEST],
+      [{ arg: null }, 0], // should be: ERROR_BAD_REQUEST
+      [{ arg: '' }, 0], // should be: ERROR_BAD_REQUEST
+    ]);
+  });
+
+  describe('json form - integer - optional', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'integer' }, [
+      // Empty values
+      [EMPTY_BODY, undefined],
+      [{ arg: null }, 0], // should be: null
+
+      // Valid values
+      [{ arg: 0 }, 0],
+      [{ arg: 1 }, 1],
+      [{ arg: -1 }, -1],
+
+      // Integers larger than MAX_SAFE_INTEGER get trimmed
+      [{ arg: 2343546576878989879789 }, 2.34354657687899e+21],
+      [{ arg: -2343546576878989879789 }, -2.34354657687899e+21],
+
+      // Scientific notation works
+      [{ arg: 1.234e+30 }, 1.234e+30],
+      [{ arg: -1.234e+30 }, -1.234e+30],
+
+      // Integer-like string values should not trigger ERROR_BAD_REQUEST
```

Comment:
Ideally, I would expect that JSON request body `{ arg: '0' }` triggers 400 Bad request when `arg` is expected to be a number. The test suite for numbers ([link](https://github.com/strongloop/strong-remoting/blob/27db434b7b252efbfea7128142e5b49a49eb16a5/test/rest-coercion/jsonform-number.suite.js#L53)) says "Number-like string values should trigger ERROR_BAD_REQUEST", I don't see why integers should behave differently.